api/hasura/actions/create_circle.ts

```diff
@@ -0,0 +1,221 @@
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+import { GraphQLClient, gql } from 'graphql-request';
+import Validator from 'validatorjs';
+
+export default async function handler(req: VercelRequest, res: VercelResponse) {
+  const variables = req.body.input.object;
+  const rules = {
+    user_name: 'required|string|max:255',
+    address: ['required', 'regex:/^(0x)?[0-9a-f]{40}$/i'],
+    circle_name: 'required|string|max:255',
+    protocol_id: 'integer',
+    protocol_name: 'required_without:protocol_id|string|max:255',
+    uxresearch_json: 'json',
+  };
+
+  Validator.register(
+    'json',
+    function (value) {
+      try {
+        JSON.parse(value);
+      } catch (e) {
+        return false;
+      }
+      return true;
+    },
+    'The :attribute is not a valid JSON.'
+  );
```

Comment:
Would be cleaner to use a library like Zod so we can get proper integration with TS for type guarding and more reusable parsers / validators from the composability.

Reply:
Yes, using zod is preferred, we can share validation code between the front end and backend.

For that, we can have a folder in `lib` for zod schemas, and actually there is one for this purpose in `src/forms/CreateCircleForm.tsx`

you might need to split it into parts. Also we can remove uxresearch_json, there is a ticket to remove those fields.

Reply:
by split into parts I mean:
```
const shared = {   user_name: z.string(), circle_name: z.string() };
const schema = { ...shared, captcha_token: z.string() }
```